src/services/study/network-map.js

```diff
@@ -216,6 +249,64 @@ export async function createMapFilter(
         distDir.id?.toString() ?? ''
     );
 }
+export async function createMapContingencyList(
+    contingencyList,
+    distDir,
+    studyUuid,
+    currentNodeUuid,
+    selectedEquipments
+) {
+    let equipmentContingencyList = [];
+    switch (contingencyList.equipmentType) {
+        case EQUIPMENT_TYPES.SUBSTATION:
+        case EQUIPMENT_TYPES.LINE:
+            equipmentContingencyList = createIdentifierContingencyList(
+                contingencyList.name,
+                selectedEquipments
+            );
+
+            break;
+
+        default:
+            if (selectedEquipments.length === 0) {
+                throw new Error('EmptySelection');
+            }
+
+            const selectedEquipmentsIds = selectedEquipments.map(
+                (element) => element.id
+            );
+            const elementsIds = await fetchNetworkElementsInfos(
+                studyUuid,
+                currentNodeUuid,
+                selectedEquipmentsIds,
+                contingencyList.equipmentType,
+                EQUIPMENT_INFOS_TYPES.MAP.type,
```

Comment:
you should probably use TAB instead of map, because Map is not implemented for Generator, Load, HVDC Converter station
500 Internal Server Error, message : 500 : "{"timestamp":"2024-05-06T08:22:17.533+00:00","status":500,"error":"Internal Server Error","message":"TODO","path":"/v1/networks/f4a533fc-0229-4a97-ad90-87d0eaef223e/elements

Reply:
some Mapper implement FORM only and others TAB, i don' know what is the best solution, may be to have a map for each equipement {EquipementType, InfoType}

see org/gridsuite/network/map/dto/mapper/

Reply:
I implemented List for all equipments !